primitives/src/lib.rs

```diff
@@ -33,15 +34,18 @@ pub type Price = FixedU128;
 /// Max fraction of pool to buy in single transaction
 pub const MAX_OUT_RATIO: u128 = 3;
 
-// Max fraction of pool to sell in single transaction
+/// Max fraction of pool to sell in single transaction
 pub const MAX_IN_RATIO: u128 = 3;
 
+/// Trading limit
+pub const MIN_TRADING_LIMIT: Balance = 1000;
+
 /// Scaled Unsigned of Balance
 pub type HighPrecisionBalance = U256;
 pub type LowPrecisionBalance = u128;
 
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq)]
+#[derive(Debug, Encode, Decode, Clone, Copy, PartialEq, Eq)]
```

Comment:
did we use this in the end?

Reply:
Yes,  It reduces the clones in the code as you don't need explicitly call it.  So it helps here with readability mainly too. In theory,  may be less expensive to perform.